phases/09-reinforcement-learning/06-policy-gradients-reinforce/code/main.py

```
@@ -0,0 +1,155 @@
+
+def reinforce(episodes, lr=0.05, gamma=0.99, use_baseline=False, rng=None):
+    rng = rng or random.Random(0)
+    theta = init_theta(rng)
+    baseline = 0.0
+    returns_log = []
+    for ep in range(episodes):
+        traj = rollout(theta, rng)
+        returns = returns_to_go(traj, gamma)
+        if use_baseline:
+            baseline = 0.95 * baseline + 0.05 * returns[0]
+        for (x, a, _r, probs), G in zip(traj, returns):
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)